fplll/pruner/pruner.h

```diff
@@ -290,47 +390,207 @@ template <class FT> class Pruner
       symmetry_factor = 1;
     }
     min_pruning_coefficients.resize(d);
+    btmp.resize(d);
+    bftmp.resize(n);
     fill(min_pruning_coefficients.begin(), min_pruning_coefficients.end(), 0.);
-
     set_tabulated_consts();
+    // need to fix target if possible
+    if (metric == PRUNER_METRIC_PROBABILITY_OF_SHORTEST)
+    {
+      // if target > 1 or < 0, optimize overall cost, use
+      //   0.99 as target for the computation in repeated_enum_cost().
+      if (this->target > 1.0 || this->target < 0.0)
+      {
+        this->target = 0.99;
+        opt_overall  = true;
+      }
+    }
+    else if (metric == PRUNER_METRIC_EXPECTED_SOLUTIONS)
+    {
+      // if target < 0, optimize overall cost, use
+      //   0.99 as target for the computation in repeated_enum_cost().
+      if (this->target < 0.0)
+      {
+        this->target = 0.99;
+        opt_overall  = true;
+      }
+    }
+    else
+    {
+      throw std::invalid_argument("Pruner was set to an unknown metric");
+    }
     load_basis_shapes(gso_rs);
   }
 
   /**
+     @brief run the optimization process using 'even' coefficients. Note
+     the optimization only applies to the pruning coefficients indexed
+     by (0, 2, 4, ... n). It thus uses half of the coefficients.
+
+     Run the optimization process, successively using the algorithm activated
+     using using half coefficients.
+  */
+  void optimize_coefficients_evec(/*io*/ vector<double> &pr);
```

Comment:
Is the documentation up to date, i.e. is pr supposed to be a vector of length `n/2` or of length `n`? The `Pruner` class gets a flag for half vs full, so I don't see why there's an additional choice of public function to select which one to use?

Reply:
I updated it. The input pr is of length n; but the function optimize_coefficients_evec(pr) will only use the even indices among them (this is to be compatible with previous version). 

Yes, the flag to the Pruner class decides whether it is full or half. If it is half, it will call optimize_coefficients_evec(pr).